packages/svelte-query/src/__tests__/createQuery.test.ts

```diff
@@ -17,14 +17,12 @@ describe('createQuery', () => {
       },
     })
 
-    expect(screen.queryByText('Loading')).toBeInTheDocument()
-    expect(screen.queryByText('Error')).not.toBeInTheDocument()
-    expect(screen.queryByText('Success')).not.toBeInTheDocument()
-
-    await sleep(20)
+    await waitFor(() => {
+      expect(rendered.getByText('Loading')).toBeInTheDocument()
+    })
 
-    expect(screen.queryByText('Success')).toBeInTheDocument()
-    expect(screen.queryByText('Loading')).not.toBeInTheDocument()
-    expect(screen.queryByText('Error')).not.toBeInTheDocument()
+    await waitFor(() => {
+      expect(rendered.getByText('Success')).toBeInTheDocument()
+    })
```

Comment:
@lachlancollins FYI, I've seen this test fail randomly on CI. It's usually better to write tests that do not rely on timing (`await sleep(20)`). I've re-written the test to match how we'd like all other tests to look as well. I'm aware many of the tests in react-query also need to catch-up 😅

Reply:
Ahh I see, thanks!